compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/DecodeV1.hs

```diff
@@ -119,11 +119,18 @@ decodeDottedName :: Util.EitherLike LF1.DottedName Int32 e
 decodeDottedName wrapDottedName mbDottedNameOrId = mayDecode "dottedName" mbDottedNameOrId $ \dottedNameOrId -> do
     (_, unmangledOrErr) <- case Util.toEither dottedNameOrId of
         Left (LF1.DottedName mangledV) -> decodeInternableStrings mangledV 0
-        Right dnId -> decodeInternableStrings V.empty dnId
+        Right dnId -> lookupDottedName dnId
     case unmangledOrErr of
       Left err -> throwError $ ParseError err
       Right unmangled -> pure $ wrapDottedName (coerce unmangled)
 
+decodeDottedNameId :: ([T.Text] -> a) -> Int32 -> Decode a
+decodeDottedNameId wrapDottedName dnId = do
```

Comment:
I might be guilty of dodging to write this function in the past and having used `decodeDottedName wrapDottedName (Just (Right dnId))` instead. Thanks for doing it properly!